packages/core/src/data-module/data-cache/caching/caching.ts

```diff
@@ -104,6 +106,94 @@ export const getDateRangesToRequest = ({
     .map(([startMS, endMS]) => [new Date(startMS), new Date(endMS)] as [Date, Date]);
 };
 
+/**
+ * Returns all the request information required
+ * Returns empty list if there are no date ranges needed to be requested.
+ *
+ * This takes into account what date intervals for a given stream id and resolution exist,
+ * allowing us to only request what is needed.
+ *
+ * It also includes all the request behaviour pertaining to each date range.
+ */
+export const getRequestInformations = ({
+  request,
+  store,
+  dataStreamId,
+  start,
+  end,
+  resolution,
+  cacheSettings,
+}: {
+  request: TimeSeriesDataRequest;
+  store: DataStreamsStore;
+  dataStreamId: string;
+  start: Date;
+  end: Date;
+  resolution: string;
+  cacheSettings: CacheSettings;
+}): RequestInformationAndRange[] => {
+  // get sorted date ranges that need to be requested
+  const dateRanges = getDateRangesToRequest({
+    store,
+    dataStreamId,
+    start,
+    end,
+    resolution: parseDuration(resolution),
+    cacheSettings,
+  });
+
+  // convert date ranges to request information
+  const requestInformations: RequestInformationAndRange[] = dateRanges.map(([rangeStart, rangeEnd], index) => ({
+    start: rangeStart,
+    end: rangeEnd,
+    id: dataStreamId,
+    resolution,
+    fetchFromStartToEnd: request.settings?.fetchFromStartToEnd,
```

Comment:
with the current logic, if `fetchFromStartToEnd` is false, then none of these request informations does anything - we should make sure we don't create `requestInformations` that 'fetch nothing'